### PR TITLE
Stats front

### PR DESCRIPTION
**Pull request by maujogue**
### Stats dashboard 
It's not polished or finished, but it should be working for now, I'll dive more into it if we have time.

- The dashboard should be blocked to users who have not made a match yet. Then you can see all the stats.
- You can change the chart mode by clicking on the respective dropdowns.

- I also changed the isLoggedIn function to use only the sessionID.
- the runEndPoint function now uses a second required argument : Method, which is "GET"/ "POST". Third one (body), is still optional

### Router Rework
- currently, every page was reloaded each time you went on it, which was defeating the core purpose of the SPA. So now, each is initialised in a separate hidden div, then when you click on it the router intelligently hides the current page and reveals the new one, so it's faster. I re-initialise all pages on login and logout, so that there is no conflict in the content.
- However, this means that if you want to refresh the content of a page, you have to do it manually using the `updatePage` or `updateModule` respectively. 
For, example, at the end of a game, I call updateModule("statisticsModule") so that the stats on the dashboard are updated with the latest match. Be aware.
- This lead to issues with the game. Especially : when re-initialising the pages (on login/logout), the event listeners for click and keyboard input would duplicate. So I changed the listeners so that they apply to the `game div` and not anymore to the `document`. This means you have to click on the game div to have it in focus and to allow the inputs.
I set the opacity of the div so that you can see when it's not in focus, there's surely a more explicit way of doing it if it's necessary.

### Fixes and miscellaneous: 
- register form being already filled when logging out
- username and email inputs glitching after changing them
- Makefile Armageddons stopping if one command can't be fulfilled
- Makefile new rule `make nomail` to disable the emailVerification for tests purposes using the python script at the root of the project
- loading spinner when the pages are being loaded, updated.

For technical reasons, I changed all `frontend/js/pages` from uppercase to lowercase, so if you have issues when checking out/cloning this branch, please check those.

Please test this thoroughly!
Thanks :)

### CODE REVIEW

**Review by boulatrousse — CHANGES_REQUESTED** [review]
Everything looks good as far as the stats themselves are concerned. I played 2 matches online and the display was correct.


However, I sometimes had a strange display. When you're on Dashboard with 0 matches played, sometimes you get this display : 
image

other times you don't and in the last case, the display stays on for 0.5 second then disappears.

Another point I noticed: when you refresh the page while you're connected, during the reload you see the home page as if you weren't connected for 1 second, then the correct display appears.

**Review by LezardSC — DISMISSED** [review]
-A little slow after refreshing and weird moving effect for the bars
-exchanges stats doesn't make any sense in these formes
-Sometimes when I go to dashboard it shows nothing and everything is at 0, then I go back to game then dashboard again and this time it works
-win streak should be renamed in 'best win streak' if it's not the current win streak. You could make a 'best' and 'current' win streak stats that could be cool

I tried to do a lot of games to see if there was a scroll bar or something and it is and it seems to works well so that's good

**Comment by gclement42** [comment]
When I tried to connect with 42 auth, after authorize I have an error message "Failed to obtain access token from 42 api"

File: py_backend/users/views/check_user_logged_in.py [hunk 1]
```diff
@@ -6,26 +6,7 @@
...
-    return False
-
-@require_http_methods(["POST"])
+@require_http_methods(["GET"])
 @requires_csrf_token
-def check_user_logged_in_view(request, username):
-    is_logged_in = is_user_logged_in(username)
-    return JsonResponse({'is_logged_in': is_logged_in})
+def check_user_logged_in_view(request):
+    return JsonResponse({'is_logged_in': request.user.is_authenticated})
```
Comment (gclement42) [inline py_backend/users/views/check_user_logged_in.py]: A views must be return a code error, if the user is not logged in return 401 otherwise return 200

**Comment by gclement42** [comment]
Globally it's good, the icon loading it's a good idea, and disabled input if div game is not focus it's also a good practice.
Otherwise
- for me the winrate graph is not suitable
- the view "check logged in" doesn't return code error 
- I have an error with auth 42. see(#97)

**Comment by maujogue** [comment]
> Globally it's good, the icon loading it's a good idea, and disabled input if div game is not focus it's also a good practice. Otherwise
> * for me the winrate graph is not suitable

True, I'll change it later on when We finalise the design of the website, I'm focusing on functionality for now.
> * the view "check logged in" doesn't return code error
Should it ? 

> * I have an error with auth 42. see([connect with 42 auth #97](https://github.com/maujogue/transcendence/issues/97))
I should remind you the secret key is not pushed to github anymore, so the 42auth will not work until you put manually in the env.prod file

**Review by gclement42 — APPROVED** [review]

**Review by boulatrousse — APPROVED** [review]
nice job, lgtm

**Pull request merged**